raft/leaderElection/leaderelection.go

```diff
@@ -11,11 +11,23 @@ import (
 
 // LeaderElection implements Raft leader election
 func LeaderElection(config types.Configuration, IP string) {
-	// for !findALeader(config, IP) {
-	findALeader(config, IP)
-	log.Println("No leader found, repeating election")
-	// }
-	log.Println("Leader found!")
+	incrementTermAndBecomeCandidate(IP)
+	log.Printf("%s incremented term to %d\n", IP, types.ServerData[IP].CurrentTerm)
+
+	rand.Seed(time.Now().UnixNano())
+	timer := getTimer(150, 300)
+	log.Printf("%s set timeout at %d\n", IP, timeOut)
+
+	voteChan := make(chan int)
+	requestvotes.RequestVotes(config, IP, timer, voteChan)
+	votes := <-voteChan
+	log.Printf("Votes for %s is %d\n", IP, votes)
+
+	if majorityObtained(votes, config) {
+		log.Println("Leader found!")
+	} else {
+		log.Println("No leader found, repeating election")
```

Comment:
ding

I think no leader found isn't the right message. Isn't it just that that particular node isn't a leader.

Reply:
Having no message would be better - less clutter in logs.